src/standalone/Vegetation/soil_drivers.jl

```diff
@@ -0,0 +1,47 @@
+using ..ClimaLSM.Soil
+export AbstractSoilDriver,
+    PrescribedSoil, PrognosticSoilHydrology, PrognosticSoil
+
+"""
+     An abstract type of soil drivers of the canopy model.
+"""
+abstract type AbstractSoilDriver{FT <: AbstractFloat} end
+
+"""
+     PrescribedSoil{} <: AbstractSoilDriver
+
+A container for holding prescribed soil parameters needed by the canopy model
+when running the canopy in standalone mode, including the soil pressure and 
+albedo.
+$(DocStringExtensions.FIELDS)
+"""
+struct PrescribedSoil{FT} <: AbstractSoilDriver{FT}
+    "The depth of the root tips, in meters"
+    root_depths::Vector{FT}
+    "Prescribed soil potential (m) as a function of time"
+    ψ_soil::Function
+    "Soil albedo"
+    soil_α::FT
+end
+
+"""
+     PrognosticSoilHydrology{FT} <: AbstractSoilDriver
+
+concrete type of AbstractSoilDriver used for dispatch in cases where a canopy 
+model and soil model are integrated hydraulically, but where soil albedo is not 
+included as a field of the soil hydraulics model.
+"""
+struct PrognosticSoilHydrology{FT} <: AbstractSoilDriver{FT}
+    soil_α::FT
+end
+
+"""
+   PrognosticSoil{FT} <: AbstractSoilDriver{FT}
+
+Concrete type of AbstractSoilDriver used for dispatch in cases where both 
+a canopy model and soil model are run. Conains a pointer to the integrated 
+soil model to allow canopy components to access the soil parameters
+"""
+struct PrognosticSoil{FT} <: AbstractSoilDriver{FT}
+    PrognosticSoilModel::AbstractSoilModel{FT}
```

Comment:
just a thought.
we dont necessarily need to store the soil model here, I think. We could perhaps just store the albedo? The instance is made in the soil_canopy constructor so we can extract soil.parameters.albedo there to make it.

Reply:
Yes you're right we can do that and then we can manipulate that definition to still allow us to use the SoilPlantHydrology model internally as well